### DISCUSSION TITLE
Problem of to select all text and reverse order of pasting copied text conversation

### DISCUSSION THREAD

**Discussion by Richard7656** [oobabooga/text-generation-webui#3436] [Q&A] [answered]
In Window 11, after using text-generation-webui to generate a few  text conversation in chat or instruct mode, it is difficult to selected all the conversation freely except use ctrl + A to selected all text in the web page. When I copy all text and paste them to Word, Writer or other software used to process text, the order of conversation are all reversed. No matter using Microsoft Edge, Google Chrome or Firefox, same problem still appear. On the other hand, this problem has not appear in GPT4All and it provide a extra button for user to copy all conversation as plain text and paste in correct order. Please check and fix the bug or error. Thank you!

Follow will show the example of the error  mention above.

### Original text conversation show in text-generation-webui :
![螢幕擷取畫面 2023-08-03 223457](https://github.com/oobabooga/text-generation-webui/assets/141255885/d2a4460a-46b7-4ba5-aa92-444789640031)

### The reverse order of text conversation paste in Writer:
![螢幕擷取畫面 2023-08-03 234305](https://github.com/oobabooga/text-generation-webui/assets/141255885/7c4632e5-9ec1-4e2c-b3cb-445fc6e6b20f)

**Comment by viperwasp** [comment]
Yes I've known of this issue for months. There is a way to save conversations however it does save as a sort of format that is a tiny bit difficult to read but at least it's in order. Possible solutions. 

1. Re-program the main chat window so ctrl+A does copy the whole conversation in order properly. 
2. Add a secondary download option that saves the conversation in a clean way. No //n/ type stuff.
3. Make it so the current way of saving conversations is a little easier to read but this may interfere with the ability to properly reload previous conversations.  

Likely option 1 or 2 would be best. But I am no programmer I don't know how to properly request or suggest this. So it's on everyone else.

**Reply by missionfloyd** [reply, marked as answer]
You can use Print (Ctrl+P) to save it as a PDF.

**Reply by viperwasp** [reply]
Thank you I will give a try next time. Might be very useful.

**Comment by oobabooga** [comment]
This happens due to this style defined in `css/chat.css`:

```css
.chat {
    display: flex;
    flex-direction: column-reverse;
}
```

It keeps the chat box scrolled to the bottom at all times, but with the side effect of inverting the copied and pasted text.

If you can find another way to do it, I would be happy to know.

**Reply by viperwasp** [reply]
With my minimal programming knowledge the only thing I can think of is a copy to clipboard button that also reverses it in the clipboard or as it's copied? Perhaps the click to copy to clipboard button can bypass the flex- direction or apply it twice? lol 

Yes I don't know if that logic even plays out or is possible. I already use the built in save chat function to save my chats. While it's not the cleanest solution it's the only one I am aware of. So maybe others can weigh in. Thanks though for providing this information Oobabooga!
